features/step_definitions/journey/registration_steps.rb

```diff
@@ -169,3 +101,100 @@
+  @reg_balance = 154
+
+  step("I want to register as an upper tier carrier")
+  step("I start a new registration journey in 'England' as a 'soleTrader'")
+  step("I complete my registration")
+  step("I pay by card")
+
+  @reg_number = @journey.confirmation_page.registration_number.text
+  puts "Registration " + @reg_number + " completed"
+end
+
+Given("a registration with outstanding balance and {int} copy cards has been submitted") do |copy_cards|
+  # Store variables for later steps:
+  @copy_cards = copy_cards
+  @reg_balance = 154 + 5 * copy_cards
+
+  step("I want to register as an upper tier carrier")
+  step("I start a new registration journey in 'England' as a 'soleTrader'")
+  step("I complete my registration")
+  step("I pay by bank transfer")
+
+  @reg_number = @journey.confirmation_page.registration_number.text
+  puts "Registration " + @reg_number + " submitted with #{copy_cards} copy cards and outstanding balance"
+end
+
+Given("a limited company with companies house number {string} is registered as an upper tier waste carrier") do |ch_no|
+  # Store variables for later steps:
+  @business_name = "AD UT Company convictions check ltd"
+  @companies_house_number = ch_no
+
+  step("I want to register as an upper tier carrier")
+  step("I start a new registration journey in 'England' as a 'limitedCompany'")
+  step("I complete my registration")
+  step("I pay by card")
+
+  @reg_number = @journey.confirmation_page.registration_number.text
+  puts "Registration " + @reg_number + " completed with conviction match on company number"
+end
+
+Given("a key person with a conviction registers as a sole trader upper tier waste carrier") do
+  # Store variables for later steps:
+  @business_name = "AD UT Sole Trader"
+  @people = @back_app.key_people_page.dodgy_people
+
+  step("I want to register as an upper tier carrier")
+  step("I start a new registration journey in 'England' as a 'soleTrader'")
+  step("I complete my registration")
+  step("I pay by card")
+
+  @reg_number = @journey.confirmation_page.registration_number.text
+  puts "Registration " + @reg_number + " completed with conviction match on relevant person"
+end
+
+Given("a conviction is declared when registering their partnership for an upper tier waste carrier") do
+  # Store variables for later steps:
+  @business_name = "AD Upper Tier Partnership"
+  @declared_convictions = :yes
+
+  step("I want to register as an upper tier carrier")
+  step("I start a new registration journey in 'England' as a 'partnership'")
+  step("I complete my registration")
+  step("I pay by card")
+
+  @reg_number = @journey.confirmation_page.registration_number.text
+  puts "Registration " + @reg_number + " completed with declared convictions"
+end
+
+Given("a registration with declared convictions is submitted with outstanding payment") do
+  # Store variables for later steps:
+  @business_name = "AD Upper Tier Need Payment"
+  @declared_convictions = :yes
+  @reg_balance = "154"
```

Comment:
I don't understand why this is a string - all other instances of `@reg_balance` are integers. I think this was carried forward from the previous PR.

Reply:
No, I added it. Because for the rest of the code it doesn't really matter if is a string or an integer. Will change that if you like

Reply:
```suggestion
  @reg_balance = 154
```